transport/peer.go

```diff
@@ -0,0 +1,48 @@
+package transport
+
+//go:generate mockgen -destination=transporttest/peer.go -package=transporttest go.uber.org/yarpc/transport PeerIdentifier,Peer
+
+// PeerConnectionStatus maintains information about the Peer's connection state
+type PeerConnectionStatus int
+
+const (
+	// PeerAvailable indicates the Peer is available for requests
+	PeerAvailable PeerConnectionStatus = iota + 1
```

Comment:
python has put the fear in me of having "falsey" values here.  But, as long as go is safe and we don't try to evaluate the connection status as a bool this should be safe.